scripts/install-deps.sh

```diff
@@ -20,76 +18,201 @@ install_deps_ubuntu()
-        packages=( ${packages[*]} libboost-filesystem-dev g++-8 )
+        packages="$packages libboost-filesystem-dev g++-8"
     fi
 
-    apt install ${packages[*]}
-    sudo snap install --classic powershell
+    case $RELEASE in
+        "18.04" | "19.04" | "20.04" | "21.04")
+            # Older Ubuntu's don't have a recent enough fmt / range-v3, so supply it.
+            fetch_and_unpack \
+                range-v3-0.11.0 \
+                range-v3-0.11.0.tar.gz \
+                https://github.com/ericniebler/range-v3/archive/refs/tags/0.11.0.tar.gz
+
+            fetch_and_unpack \
+                fmt-8.0.1 \
+                fmtlib-8.0.1.tar.gz \
+                https://github.com/fmtlib/fmt/archive/refs/tags/8.0.1.tar.gz
+            ;;
+        *)
+            packages="$packages libfmt-dev librange-v3-dev"
+            ;;
+    esac
+
+    sudo apt install $packages
+    #sudo snap install --classic powershell
 }
 
 install_deps_fedora()
 {
-    local packages=(
-	fontconfig-devel
+    # catch-devel
+    local packages="
         cmake
         extra-cmake-modules
+        fmt-devel
+        fontconfig-devel
         freetype-devel
         gcc-c++
+        gsl-devel
         harfbuzz-devel
         kf5-kwindowsystem-devel
         ninja-build
         pkgconf
         qt5-qtbase-devel
         qt5-qtbase-gui
-    )
-    sudo dnf install ${packages[*]}
+        yaml-cpp-devel
+    "
+    sudo dnf install $packages
 }
 
-main_linux()
+
+main_darwin()
 {
-    #local ID=`lsb_release --id | awk '{print $NF}'`
-    local ID=$(grep ^ID= /etc/os-release | cut -d= -f2)
+    # NB: Also available in brew: mimalloc
+    brew install \
+        boost \
+        catch2 \
+        cpp-gsl \
+        fontconfig \
+        fmt \
+        freetype \
+        harfbuzz \
+        pkg-config \
+        range-v3 \
+        yaml-cpp
+}
 
-    case "${ID}" in
-        fedora)
-            install_deps_fedora
-            ;;
-        ubuntu|neon)
-            install_deps_ubuntu
-            ;;
-        *)
-            echo "No automated installation of build dependencies available yet."
-            ;;
-    esac
+# {{{ sysdeps fetcher and unpacker for deps that aren't available via sys pkg mgnr
+GSL_VERSION="3.1.0"
+TERMBENCH_PRO_VERSION="5a79261fbf5d26c9bf9d9a0d31f22ef4556cdd3b"
+LIBUNICODE_VERSION="46a48c25df2b57501ef97fde62b126c232f5f259"
+
+SYSDEPS_BASE_DIR="$(dirname $0)/../_deps"
+
+SYSDEPS_DIST_DIR="$SYSDEPS_BASE_DIR/distfiles"
+SYSDEPS_SRC_DIR="$SYSDEPS_BASE_DIR/sources"
+SYSDEPS_CMAKE_FILE="$SYSDEPS_SRC_DIR/CMakeLists.txt"
+
+fetch_and_unpack()
+{
+    NAME=$1
+    DISTFILE=$2
+    URL=$3
+
+    FULL_DISTFILE="$SYSDEPS_DIST_DIR/$DISTFILE"
+
+    if ! test -f "$FULL_DISTFILE"; then
+        if which curl &>/dev/null; then
+            curl -L -o "$FULL_DISTFILE" "$URL"
+        elif which wget &>/dev/null; then
+            wget -O "$FULL_DISTFILE" "$URL"
+        else
+            echo "Don't know how to fetch from the internet." 1>&2
```

Comment:
For the future in case of FreeBSD, you can use `fetch(1)` which is included in the base system:

```
$ fetch <url> -o <target-file>
```

Reply:
Nice. Thanks.

Reply:
Yes. That's there plan. That this script ISD about to support FreeBSD, too.

Reply:
done.